### PR TITLE
docs: tier review round 1 by surface, drop forced minimum rounds

### LINKED ISSUE #1180: Shorten the self-review loop: tier round 1, drop forced minimum rounds
## Problem

The self-review loop is still too long after #1159 (round shape, liveness) and #1173 (substantive gate, round budget). Every PR that enters the loop, whatever it touches, pays a fixed minimum of three sequential reviewer stages:

1. **Round 1 is the deep-review workflow for EVERY task**, which costs a scout + 6 fixed lenses + up to 6 dynamic lenses + an adversarial jury per finding (7 to 24 agents), even for a docs-only or skill-file PR.
2. **A forced second round even when deep-review confirms nothing** (the two-round minimum: "a deep-review that confirms nothing substantive is rare... run one delta-shaped pass anyway"). This adds a round to exactly the cheapest, cleanest case, and it second-guesses a result the juries already adversarially verified.
3. **A mandatory standalone prose pass** after convergence, one more fresh reviewer, even though every reviewer in the loop already reports and tags prose-tier findings (the tag is in the delta template and in deep-review's `FINDINGS` schema), so the dedicated pass mostly re-collects what the converging round already surfaced.

The owner's directive: make the review cycle shorter while keeping the quality bar where it matters. The quality asymmetry the loop itself already states is the lever: "a `packages/*/src` bug ships into every end-user app", while a docs/skill/website bug does not.

## Design / approach

Keep deep-review's full adversarial depth for shipped source; stop paying it, and the two forced extra rounds, everywhere else. Four changes:

1. **Tier round 1 by surface, file-checkable, never judgment-based** (the same trick as the #1171 substantive gate: surface membership is checkable from the diff, so the agent motivated to go light cannot misjudge it). If the PR's diff touches **shipped source**, defined as `packages/*/src/**`, `packages/cli/lib/**`, or `packages/cli/templates/**`, round 1 is the deep-review workflow, unchanged. Otherwise (docs site, website, blog, skills, hooks, tests-only, examples, scaffold docs) round 1 is **one broad fresh reviewer**: the existing delta-verifier template with the full diff as its question. A light-tier PR's loop is then typically one reviewer, or a handful when it finds things. The user asking for a deep review always forces the deep tier.
2. **Drop the forced second round after a clean deep-review.** A deep-review run that confirms nothing substantive ends the substantive loop by itself: its findings were already jury-refuted, which is exactly the adversarial re-check the extra delta round was standing in for. Delta rounds still run after every round that DOES produce substantive fixes, until one finds nothing substantive. The two-round minimum is replaced by: minimum one round; fixes always get a fresh delta round.
3. **Fold the prose pass into the converging round.** The round that finds nothing substantive (or whose findings are all prose once tags are verified) IS the prose sweep: its prose-tier findings are applied without re-review (already the gate's rule) and the loop ends. The standalone mandatory prose-pass reviewer is removed. The gate's one exception stays: a prose fix that touches a substantive surface gets one delta-scoped check of that fix alone. The third-party-internals content rule stays.
4. **Trim deep-review defaults**, since it now runs only where its depth is earned: dynamic lens cap 6 to 3 (the scout's "zero is a fine answer" plus measured runs rarely earning six), default `maxAgents` 24 to 16 (still clamped 8 to 60, still overridable per run).

Everything else is untouched: liveness/watchdog rules, worktree safety, the posting mechanics (one review object per round), the refuter gate on expensive delta findings, the substantive gate and its asymmetric tag verification, the 5-round budget, BLOCKED/OVER BUDGET exits, the deferral ledger, the deferred-suites + CI read after convergence, and the when-to-skip rule.

## Implementation notes (for the implementing agent)

- Where to edit:
  - `.claude/skills/webjs-start-work/SKILL.md` `### How the loop works` (L311-L385): the round-1 bullet (L324) gains the surface tier; the two-round-minimum paragraph (L381) is replaced; step 4 (L379) loses the standalone prose pass and instead ends the loop on the converging round with its prose findings applied without re-review (keep the deferred-suites/CI parallel-run text, which lives in the same step); the last-round absolute (L383), the standalone-review entry (L385, "ALWAYS starts with a fresh deep-review run" becomes tier-respecting), `### Reporting after the loop` (L391-L403, the report template says "plus the prose pass" and the `<K>` counting rules mention it), and the two cross-references at L105 and L111 ("a round finding nothing substantive, then the prose pass").
  - `.claude/workflows/deep-review.js`: `maxAgents` default at L34 (24 to 16), the dynamic-lens caps at L110 (`maxItems: 6`), L127/L141 (`slice(0, 6)`), and the scout prompt's "ZERO to six" wording at L134; `meta.whenToUse` (L4) says "The round-1 review for any PR entering a review loop", which becomes shipped-source-tier only.
  - `.claude/hooks/route-skills.sh` L190: the code-review directive hard-codes "round 1 is the deep-review workflow" and "one bounded prose pass ends it"; update in lockstep (the comment at L182 says exactly this).
- Landmines / gotchas:
  - **Two byte-identical copies of the skill exist**: in-repo and `~/.claude/skills/webjs-start-work/SKILL.md`. Every #1159/#1173 PR verified them with `diff -q` after the change; do the same, copying in-repo onto `~/.claude`.
  - #1159 explicitly justified the uniform deep-review rule ("a uniform rule cannot be misjudged the way a by-risk bifurcation can"). This issue supersedes that at the owner's direction; the tier must stay SURFACE-based (paths in the diff) precisely so the misjudgment concern stays closed. Do not write an importance-based tier.
  - The loop's cross-references are by name, not ordinal (#1159 did this deliberately); when removing the prose pass, sweep EVERY mention (`grep -n "prose pass" SKILL.md` finds them all) or a dangling reference ships.
  - `test/hooks/route-skills.test.mjs` asserts routing behavior, not loop wording, but run it anyway (16/16) since the hook file is edited.
  - `node --check` is inapplicable to `deep-review.js` (top-level `return` is legal only in the Workflow runtime's async context, per #1173's test plan); validate by wrapping as an async function body.
- Invariants to respect: AGENTS.md invariant 11 (prose punctuation, WebJs casing) applies to all edited prose; the gate changes only WHEN the loop exits, never WHETHER a surface is reviewed (both tiers still posted and resolved on the PR); the deferred suites still run in full after convergence.
- Tests + docs surfaces: `node --test test/hooks/route-skills.test.mjs`; no `packages/*/src` change, so no docs-site/website/scaffold surface (same N/A rationale as #1159/#1173).

## Acceptance criteria

- [ ] Round 1 is deep-review only when the PR diff touches `packages/*/src/**`, `packages/cli/lib/**`, or `packages/cli/templates/**`; otherwise one broad fresh reviewer, with an explicit user ask always forcing the deep tier.
- [ ] A round 1 (either tier) that finds nothing substantive ends the substantive loop with no forced second round; a round with substantive fixes still gets a fresh delta round on the new HEAD.
- [ ] The standalone prose pass is gone; the converging round's prose-tier findings are applied without re-review, and the prose-fix-touching-substantive-surface exception survives.
- [ ] `deep-review.js` defaults: `maxAgents` 16, dynamic lenses capped at 3, `whenToUse` reflects the tier.
- [ ] The route-skills.sh code-review directive matches the new loop wording; `node --test test/hooks/route-skills.test.mjs` passes 16/16.
- [ ] `grep -n "prose pass" .claude/skills/webjs-start-work/SKILL.md` shows no stale reference to the removed standalone pass.
- [ ] Both skill copies byte-identical (`diff -q`) after the change.
- [ ] Liveness, worktree-safety, posting, refuter-gate, budget, and deferral rules are textually untouched except where they cross-referenced the prose pass or two-round minimum.

### PR DESCRIPTION
**Pull request by vivek7405**
Closes #1180

The self-review loop still charged every PR a fixed minimum of three sequential reviewer stages after #1159 and #1173: a deep-review round 1 (7 to 24 agents, whatever the PR touched), a forced second round even when deep-review confirmed nothing, and a standalone prose pass. This shortens the loop while keeping the adversarial depth where a bug actually ships into end-user apps.

## What changed

In `.claude/skills/webjs-start-work/SKILL.md`, `### How the loop works`:

- **Round 1 is tiered by the diff's paths, a path check, never a judgment call.** A diff touching shipped source (`packages/*/src/`, `packages/cli/lib/`, `packages/cli/templates/`) takes the deep-review workflow as round 1, unchanged. Every other diff (docs, website, blog, skills, hooks, tests-only, examples) takes ONE broad fresh reviewer using the delta-verifier template with the full diff as its question. A mixed diff takes the deep tier (one shipped-source path is enough), the user asking for a deep review always forces it, and a scope change that pulls shipped source into a light-tier PR re-runs round 1 at the deep tier. The tier is path-based for the same reason the substantive gate is surface-based: membership is checkable from the diff, so the agent motivated to go light cannot misjudge it. This supersedes #1159's uniform-round-1 rule at the owner's direction; the misjudgment concern that rule addressed stays closed because the tier reads paths, not importance.
- **The minimum is ONE round.** A round 1 of either tier that finds nothing substantive converges the loop by itself: on the deep tier its findings were already adversarially jury-refuted, and on the light tier nothing in the diff is shipped source, so the old two-round minimum added a round to exactly the cleanest case. A fix still always buys a fresh delta round.
- **The converging round's prose-tier findings replace the standalone prose pass.** Every reviewer already reports and tags prose findings alongside substantive ones, so the round that finds nothing substantive IS the prose sweep: its prose fixes apply without re-review and the loop ends. The gate's exception survives: a prose fix that touches a substantive surface gets one delta-scoped check of that fix alone.
- The refuter gate, liveness and watchdog rules, worktree safety, posting mechanics, the substantive gate and its asymmetric tag verification, the 5-round budget, the BLOCKED and OVER BUDGET exits, the deferral ledger, and the deferred suites plus CI read after convergence are all untouched except where they cross-referenced the removed pass or minimum.

In `.claude/workflows/deep-review.js`: default `maxAgents` drops 24 to 16 (still clamped 8 to 60, still overridable per run), dynamic lenses cap at three instead of six (scout prompt, schema `maxItems`, both slices), and `whenToUse` scopes round 1 to shipped-source diffs. The six fixed lenses, the model split, the jury sizes, and every fail-open path are unchanged.

In `.claude/hooks/route-skills.sh`: the code-review directive now states the tiered round 1 and the new exit, staying in lockstep with the skill.

In `test/hooks/review-loop-exit.test.mjs`: the guard assertions pinning the prose pass and the two-round minimum are replaced by assertions pinning the tier rule, the one-round minimum, the converging-round prose application, and the trimmed workflow defaults, so a revert of any piece reds the test.

## What a loop costs now

A docs or skill PR: one broad reviewer when clean, plus a delta round per fix. A shipped-source PR: deep-review at 16 agents, plus a delta round per fix. Nothing pays a forced extra round or a separate prose reviewer anymore.

## Test plan

- `node --test test/hooks/review-loop-exit.test.mjs test/hooks/route-skills.test.mjs`: 20/20 pass (the review-loop guard, and the router test including "every skill the hook can route to is committed in-repo").
- Counterfactual: the guard test's new assertions fail against the pre-change files (they pin `The minimum is ONE round`, the tier bullet, the 16-agent default, and the absence of the standalone prose pass in either spelling). Verified whole-file against the parent commit and hunk-level for the skill's failure-handling hunk and the hook's comment hunk, both red when reverted alone.
- `deep-review.js` validity in the async workflow context is covered by the existing guard test ("stays valid in the async workflow context").
- Both copies of the skill (in-repo and `~/.claude`) verified byte-identical with `diff -q` after the change.

## Doc surfaces

Tests: the guard tests above. Markdown / docs site / scaffold / MCP / editor plugins / marketing / dogfood apps / version bumps: N/A because nothing in `packages/*/src` or any user-facing surface changed; this edits the Claude Code skill, the committed review workflow, and the routing hook, which ship to agents working this repo, not to anything the framework publishes.

### CODE REVIEW

File: test/hooks/review-loop-exit.test.mjs [hunk 1]
```diff
@@ -39,21 +39,26 @@ test('the skill carries the substantive gate and its two-tier exit', () => {
   // and the one tagless-reviewer kind the orchestrator classifies itself.
   assert.match(skill, /arrives UNTAGGED from a reviewer whose result shape carries the tag is treated as substantive/);
   assert.match(skill, /structurally cannot tag/);
-  // The bounded prose pass and its non-round status.
-  assert.match(skill, /never counts toward the round total or the two-round minimum/);
+  // The exit: the converging round's prose findings apply without a
+  // re-review round, and there is no standalone prose pass to relitigate.
+  assert.match(skill, /apply that round's prose-tier fixes WITHOUT re-review and stop/);
+  assert.match(skill, /converging round's prose-tier findings ARE the sweep/);
+  assert.ok(!skill.includes('prose pass'), 'the standalone prose pass must not resurface in the skill');
```
Comment (vivek7405) [inline test/hooks/review-loop-exit.test.mjs]: This checks the two-word spelling only. The pre-change skill wrote two of its references hyphenated ("the prose-pass reviewer"), so reverting just the failure-handling hunk brings the removed pass back and every test here stays green. Nothing else pins the new failure-handling wording either.

File: .claude/hooks/route-skills.sh [hunk 2]
```diff
@@ -178,16 +178,17 @@ fi
 # Triggers: review the PR/diff/branch/changes, code review, look it over
 # for bugs. Reviewing every change before it is marked ready is a standing
 # expectation, and a review is a LOOP with the substantive-gate exit the
-# webjs-start-work skill defines (loop on substantive findings, then one
-# bounded prose pass). Keep this directive in lockstep with that skill's
+# webjs-start-work skill defines (loop on substantive findings; the
+# converging round's prose findings apply without re-review). Keep this
```
Comment (vivek7405) [inline .claude/hooks/route-skills.sh:182]: Reverting only this comment hunk (reintroducing "then one bounded prose pass") also leaves the guard test green, since the only negative anchor is the directive-line phrasing. The PR body claims a revert of any piece reds the test, which this contradicts.

**Review by vivek7405 — COMMENTED** [review]
Went through the whole loop section end to end with the new tier in mind. The shape holds together, but the guard test has a real observability hole, and two spots still talk like the old loop. One more thing that sits on an unchanged line so I am noting it here rather than inline: in the liveness rules (SKILL.md, the returns-without-reviewing bullet), the expected-result-shape enumeration still names only two reviewer kinds, the delta verifier and a deep-review run, while the broad round-1 reviewer is now a third kind everywhere else (step 1, the template intro, the reporting section). The one enumeration that decides whether a round happened should name it too.

File: .claude/skills/webjs-start-work/SKILL.md [hunk 3]
```diff
@@ -392,15 +392,15 @@ Skip only for PRs that change a single line of trivially-correct content (a doc
 
 After the loop converges and the deferred suites plus the CI read (step 4) are done, report exactly this shape to the user:
 
-> PR #<N> is up at <URL>. Self-review loop ran <K> rounds plus the prose pass; last round found nothing substantive. Issues found and fixed during the loop: <one-line list, or "none" if rounds 2+ kept finding nothing>. Out-of-scope findings, recorded on the PR and awaiting your call on filing: <one-line list, or "none">. Ready to merge.
+> PR #<N> is up at <URL>. Self-review loop ran <K> rounds; the last found nothing substantive. Issues found and fixed during the loop: <one-line list, or "none" if rounds 2+ kept finding nothing>. Out-of-scope findings, recorded on the PR and awaiting your call on filing: <one-line list, or "none">. Ready to merge.
```
Comment (vivek7405) [inline .claude/skills/webjs-start-work/SKILL.md]: The qualifier still says "none if rounds 2+ kept finding nothing", which presumes rounds 2+ exist. Under the one-round minimum a converged loop can be exactly one round, and the condition then describes nothing.

File: test/hooks/review-loop-exit.test.mjs (hunk 1 above)
Comment (vivek7405) [reply test/hooks/review-loop-exit.test.mjs]: Fixed in [`0b840df5`](https://github.com/webjsdev/webjs/commit/0b840df5).

File: .claude/hooks/route-skills.sh (hunk 2 above)
Comment (vivek7405) [reply .claude/hooks/route-skills.sh:182]: Fixed in [`0b840df5`](https://github.com/webjsdev/webjs/commit/0b840df5).

File: .claude/skills/webjs-start-work/SKILL.md (hunk 3 above)
Comment (vivek7405) [reply .claude/skills/webjs-start-work/SKILL.md]: Fixed in [`0b840df5`](https://github.com/webjsdev/webjs/commit/0b840df5).

**Review by vivek7405 — COMMENTED** [review]
Re-checked the observability fix and its blast radius across the whole PR surface: the tightened anchors do not false-fail anywhere, the pinned failure-handling wording is literal, the reviewer-kind enumerations agree everywhere, and nothing else still speaks in the old loop shape. Nothing to flag.
